src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.cs

```diff
@@ -9,7 +9,7 @@
 #if (NativeAot)
 builder.Services.ConfigureHttpJsonOptions(options =>
 {
-    options.SerializerOptions.AddContext<AppJsonSerializerContext>();
+    options.SerializerOptions.TypeInfoResolverChain.Insert(0, AppJsonSerializerContext.Default);
```

Comment:
It isn't really about the "default" type resolver to use though. For example if you call `AddProblemDetails()`, you don't want to use this type resolver for the ProblemDetails types.

I assume you'd be OK with the readability of:

```C#
options.SerializerOptions.TypeInfoResolverChain.Add(AppJsonSerializerContext.Default);
```

But that's not the semantics of what you want. You really want to "prepend" by default. So if we were to add something to the framework, I think it would be a helper to add a context to the beginning of the chain.

> could we perhaps create an extension method on Microsoft.AspNetCore.Http.Json.JsonOptions

If we want something in the framework, I believe it would be better to put it in System.Text.Json itself.

Reply:
Other alternatives here are:

1. Use Combine with the new one first:
```C#
options.SerializerOptions.TypeInfoResolver =
    JsonTypeInfoResolver.Combine(AppJsonSerializerContext.Default, options.SerializerOptions.TypeInfoResolver);
```
2. Since this is only for NativeAOT, we could just use `.Add` and rely on the fact that when in NativeAOT there won't be a reflection based resolver. However, when you aren't using NativeAOT, you don't want to use `.Add` because that will put your context _after_ the reflection based resolver, which means yours won't be used. So I wouldn't go with this approach.

Reply:
My thought process was that setting the "default" type resolver in the chain means putting it first. We can leave this as is for now and we can consider if there's a higher level API that could help with the usability here later.

Reply:
Would the "Combine" API be easier on the eyes? They are equivalent.

Reply:
Possibly, although it has the problem of being quite long. Let's leave the `Insert(0` for now and revisit.

Reply:
I was worried using `JsonTypeInfoResolver.Combine` would collapse the `TypeInfoResolverChain` giving it a length of one making it less inspectable, but then I looked at the `AddFlattened` logic and saw that it should indeed be equivalent. I agree it `Combine` looks nicer than `Insert(0`.